### PR TITLE
OPENEUROPA-544: Port task runner to use latest Robo version

### PR DESCRIPTION
**Pull request by drupol**

### CODE REVIEW

File: bin/run [hunk 1]
```diff
@@ -4,12 +4,21 @@
...
 
-if (file_exists(__DIR__.'/../vendor/autoload.php')) {
-    $classLoader = require __DIR__.'/../vendor/autoload.php';
-} elseif (file_exists(__DIR__.'/../../../autoload.php')) {
-    $classLoader = require __DIR__ . '/../../../autoload.php';
+// If we're running from phar load the phar autoload file.
+$pharPath = \Phar::running(true);
+if ($pharPath) {
+    $autoloaderPath = "$pharPath/vendor/autoload.php";
+} else {
```
Comment (ademarco) [inline bin/run]: Out of scope from the current PR, serving it as a Phar has implications we need to consider before having this in.

File: composer.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "minimum-stability": "dev",
     "prefer-stable": true,
     "require": {
-        "consolidation/robo": "^1.2",
+        "consolidation/robo": "^1.3",
```
Comment (ademarco) [inline composer.json]: Setting Robo to `^1.2.2` should be enough.

File: src/Robo/Plugin/AbstractCommands.php [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace OpenEuropa\TaskRunner\Commands;
+namespace OpenEuropa\TaskRunner\Robo\Plugin;
```
Comment (ademarco) [inline src/Robo/Plugin/AbstractCommands.php]: We don't have to use Robo namespace but our own.

File: src/Robo/Plugin/AbstractCommands.php [hunk 4]
```diff
@@ -10,14 +10,15 @@
...
+use Robo\Tasks;
 use Symfony\Component\Console\Event\ConsoleCommandEvent;
 
 /**
  * Class AbstractCommands
  *
  * @package OpenEuropa\TaskRunner\Commands
  */
-abstract class AbstractCommands implements BuilderAwareInterface, IOAwareInterface, ConfigAwareInterface
+abstract class AbstractCommands extends Tasks implements BuilderAwareInterface, IOAwareInterface, ConfigAwareInterface
```
Comment (ademarco) [inline src/Robo/Plugin/AbstractCommands.php]: Any reason why we should extend `Tasks` here?

File: src/Tasks/CollectionFactory/CollectionFactory.php [hunk 5]
```diff
@@ -20,7 +20,7 @@
  *
  * Return a task collection given its array representation.
  *
- * @package OpenEuropa\TaskRunner\Tasks\YamlTaskFacorty
+ * @package OpenEuropa\TaskRunner\Tasks\YamlTaskFactory
```
Comment (ademarco) [inline src/Tasks/CollectionFactory/CollectionFactory.php]: Thanks for this!

**Comment by drupol** [comment]
Closed in favor of #61

**Pull request closed without merging**